File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/felles/OpprettEllerOppdaterResultat.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.felles;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class OpprettEllerOppdaterResultat {

    private int antallRadOpprettet;
    private int antallRadOppdatert;

    public OpprettEllerOppdaterResultat add(OpprettEllerOppdaterResultat resultat) {
        antallRadOpprettet = antallRadOpprettet + resultat.getAntallRadOpprettet();
        antallRadOppdatert = antallRadOppdatert + resultat.getAntallRadOppdatert();
        return this;
    }
}

File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/applikasjon/domenemodeller/OpprettEllerOppdaterResultat.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domenemodeller

data class OpprettEllerOppdaterResultat(
    val antallRadOpprettet: Int = 0,
    val antallRadOppdatert: Int = 0
) {
    fun add(other: OpprettEllerOppdaterResultat) = OpprettEllerOppdaterResultat(
        antallRadOpprettet + other.antallRadOpprettet,
        antallRadOppdatert + other.antallRadOppdatert
    )
}
